app/src/main/java/com/kamron/pogoiv/Pokefly.java

```diff
@@ -1186,24 +1186,34 @@ private int levelToSeekbarProgress(double level) {
      * pokemon evolution and level set by the user.
      */
     private void populateAdvancedInformation(IVScanResult ivScanResult) {
+
         double goalLevel = seekbarProgressToLevel(expandedLevelSeekbar.getProgress());
-        int intSelectedPokemon =
+        int spinnerSelection =
                 extendedEvolutionSpinner.getSelectedItemPosition(); //which pokemon is selected in the spinner
         ArrayList<Pokemon> evolutionLine = pokeInfoCalculator.getEvolutionLine(ivScanResult.pokemon);
+        extendedEvolutionSpinnerAdapter.updatePokemonList(evolutionLine);
+
+        Pokemon selectedPokemon = ivScanResult.pokemon;
 
-        Pokemon selectedPokemon;
-        if (intSelectedPokemon == -1) {
-            selectedPokemon = ivScanResult.pokemon;//if initialising list, act as if scanned pokemon is marked
-            for (int i = 0; i < evolutionLine.size(); i++) {
-                if (evolutionLine.get(i).number == selectedPokemon.number) {
-                    intSelectedPokemon = i;
-                    extendedEvolutionSpinner.setSelection(intSelectedPokemon);
+        if (spinnerSelection == -1) { // initializing the estimation box
+            boolean hasEvolution = selectedPokemon.evolutions.size() > 0;
+            int selectIndex = 0;
+            for (Pokemon evoPoke : evolutionLine) { //find the index of the pokemon in evolution line
+                if (evoPoke.number == selectedPokemon.number) {
                     break;
                 }
+                selectIndex++;
             }
+            if (hasEvolution) { //select the next pokemon in the spinner if evolution exists
+                selectIndex++;
+            }
```

Comment:
So IIUC these three lines are the actual bugfix, and everything else is just a refactoring, right? Next time, it'd be nice to separate those changes, it's really hard to review them together.